src/charm.py

```diff
@@ -195,17 +217,23 @@ def remove_secret(self, scope, key) -> None:
         content[key] = Config.Secrets.SECRET_DELETED_LABEL
         secret.set_content(content)
 
+    def stop_mongos_service(self):
+        """Stop mongos service."""
+        container = self.unit.get_container(Config.CONTAINER_NAME)
+        container.stop(Config.SERVICE_NAME)
+
     def restart_charm_services(self):
         """Restart mongos service."""
         container = self.unit.get_container(Config.CONTAINER_NAME)
-        container.stop(Config.SERVICE_NAME)
+        try:
+            container.stop(Config.SERVICE_NAME)
+        except APIError:
```

Comment:
An `APIError` or a `ChangeError`?
Why ignore? How can we restart a service if the stop failed?

Reply:
please read comment below

Reply:
My question is with regards to the type of the caught exception. 
I cannot see in the docstring an error in the stop process raising an `APIError`. I see however `ChangeError`.

Reply:
weird, when I run it by hand I get `APIError`, not change error, I can report a bug